xarray/backends/file_manager.py

```diff
@@ -0,0 +1,121 @@
+import contextlib
+import threading
+
+from .lru_cache import LRUCache
+
+
+# Global cache for storing open files.
+FILE_CACHE = LRUCache(512, on_evict=lambda k, v: v.close())
+
+# TODO(shoyer): add an option (xarray.set_options) for resizing the cache.
+# Note: the cache has a minimum size of one.
+
+
+class FileManager(object):
+    """Wrapper for automatically opening and closing file objects.
+
+    Unlike files, FileManager objects can be safely pickled and passed between
+    processes. They should be explicitly closed to release resources, but
+    a per-process least-recently-used cache for open files ensures that you can
+    safely create arbitrarily large numbers of FileManager objects.
+
+    Example usage:
+
+        manager = FileManager(open, 'example.txt', mode='w')
+        with manager.acquire() as f:
+            f.write(...)
+        manager.close()
+    """
+
+    def __init__(self, opener, *args, **kwargs):
```

Comment:
I think I want to use dependency injecting for the cache (e.g., `cache=FILE_CACHE`), which unfortunately means that we'll need to change the signature here from using `**kwargs`.

Any opinions on what this should look like? I'm thinking maybe:
```python
_DEFAULT = object()
def __init__(self, opener, *args, mode=_DEFAULT, kwargs=None, cache=FILE_CACHE)
```